src/hardware/input/intel8042.cpp

```diff
@@ -371,6 +371,63 @@ static uint8_t get_translated(const uint8_t byte)
 	return translation_table[byte];
 }
 
+// ***************************************************************************
+// Port disable workaround
+// ***************************************************************************
+
+// TODO: This part is a workaround for 'Ultima VIII: Pagan' problem with
+// non-working keyboard. The proper fix would require larger rework:
+// - it seems that even if keyboard is disabled, the 8042 should fire the
+//   interrupt
+// - testing with 86Box revealed that command 0xad (disable keyboard port) and
+//   0xae (enable port) are being sent by BIOS with each and every key press;
+//   our internal BIOS does not work this way
+//
+// Ignoring command 0xad totally might be risky, as Windows 3.11 for Workgroups
+// disables/reenables the keyboard port during startup and shutdown, issuing
+// controller commands in between; if something disturbs it, we might run into
+// a problem with keyboard/mouse not responding at all.
+//
+// This code re-enables the keyboard after certain time - unless there is
+// some command currently being issued to the controller.
```

Comment:
Can you think of any cases where this re-enabling workaround would introduce unwanted behaviour? Is there anything we can help regression test?

Reply:
I can only think about starting/quitting different versions of Windows (from 1.0 to 9x) and see whether keyboard is still working.